File: app/src/main/java/com/moa/rxdemo/mvp/view/SplashActivity.java
package com.moa.rxdemo.mvp.view;

import android.content.Context;
import android.content.Intent;

import com.moa.baselib.base.dispatcher.Runtimes;
import com.moa.baselib.base.ui.BaseActivity;
import com.moa.rxdemo.R;

/**
 * 类或文件描述
 * <p>
 * Created by：wangjian on 2018/12/20 16:04
 */
public class SplashActivity extends BaseActivity {
    
    @Override
    protected int getLayoutId() {
        return R.layout.tt_activity_splash;
    }
    
    @Override
    protected void initData() {
        super.initData();
        Runtimes.dispatchDelay(new Runnable() {
            @Override
            public void run() {
                startActivity(MainActivity.getIntent(SplashActivity.this));
                finish();
            }
        },20);
    }
    
    public static Intent getIntent(Context context){
        Intent intent = new Intent(context, SplashActivity.class);
        return intent;
    }
}

File: app/src/main/java/com/moa/rxdemo/mvp/view/SplashActivity.kt
package com.moa.rxdemo.mvp.view

import android.content.Context
import android.content.Intent
import android.os.Handler
import com.moa.rxdemo.R
import com.moa.rxdemo.base.ui.BaseActiivty

/**
 * 类或文件描述
 *
 * Created by：wangjian on 2018/12/17 17:11
 */
class SplashActivity : BaseActiivty() {

    companion object {

        fun getIntent(context: Context): Intent {
            return Intent(context, SplashActivity::class.java);
        }
    }

    override fun getLayoutId(): Int {
        return R.layout.tt_activity_splash;
    }

    override fun initData() {
        super.initData()

        Handler().postDelayed({
            startActivity(MainActivity.getIntent(this))
            finish()
        }, 10);
    }

}